Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/retrieve/AuditRetrieveEventsUtil.java

```diff
@@ -0,0 +1,132 @@
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.List;
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBElement;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Unmarshaller;
+import javax.xml.datatype.DatatypeConfigurationException;
+import javax.xml.datatype.DatatypeFactory;
+import javax.xml.datatype.XMLGregorianCalendar;
+import javax.xml.stream.XMLStreamException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author tjafri
+ */
+public class AuditRetrieveEventsUtil {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AuditRetrieveEventsUtil.class);
+
+    public QueryAuditEventsResponseType getQueryAuditEventResponse(List<AuditRepositoryRecord> dbRecords) {
+        QueryAuditEventsResponseType response = new QueryAuditEventsResponseType();
+        if (dbRecords != null) {
+            List<QueryAuditEventsResults> auditList = new ArrayList<>();
+            QueryAuditEventsResults auditObj;
+            for (AuditRepositoryRecord rec : dbRecords) {
+                auditObj = new QueryAuditEventsResults();
+                auditObj.setDirection(rec.getDirection());
+                auditObj.setEventId(rec.getEventId());
+                auditObj.setEventOutcomeIndicator(BigInteger.valueOf(rec.getOutcome()));
+                auditObj.setEventTimestamp(convertDateToXMLGregorianCalendar(rec.getEventTimestamp()));
+                auditObj.setEventType(rec.getEventType());
+                auditObj.setId(rec.getId());
+                auditObj.setRelatesTo(rec.getRelatesTo());
+                auditObj.setRemoteHcid(rec.getRemoteHcid());
+                auditObj.setRequestMessageId(rec.getMessageId());
+                auditObj.setUserId(rec.getUserId());
+                auditList.add(auditObj);
+            }
+            response.getQueryAuditEventsResults().addAll(auditList);
+        }
+        return response;
+    }
+
+    public QueryAuditEventsBlobResponse getQueryAuditEventBlobResponse(Blob auditBlob) {
+        QueryAuditEventsBlobResponse response = new QueryAuditEventsBlobResponse();
+        if (auditBlob != null) {
+            response.setAuditMessage(unMarshallBlobToAuditMess(auditBlob));
+        }
+        return response;
+    }
+
+    private XMLGregorianCalendar convertDateToXMLGregorianCalendar(Date date) {
+        if (date != null) {
+            GregorianCalendar gregorianCalendar = new GregorianCalendar();
+            gregorianCalendar.setTime(date);
+            try {
+                return DatatypeFactory.newInstance().newXMLGregorianCalendar(gregorianCalendar);
+            } catch (DatatypeConfigurationException ex) {
+                LOG.error("Unable to convert date: " + ex.getLocalizedMessage(), ex);
+                return null;
+            }
+        }
+        return null;
+    }
+
+    private AuditMessageType unMarshallBlobToAuditMess(Blob auditBlob) {
```

Comment:
A couple "naming convention" issues:

1. Can you change to "unmarshal" (lowercase m, single l)?  
2. Consider changing "Mess" to "Msg" for consistency with other functions in this PR, e.g., https://github.com/CONNECT-Solution/CONNECT/pull/1198/files#diff-e953106289ce8c3ce877bbfff430f773R43